scripts/zones/Port_Jeuno/npcs/Sagheera.lua

```diff
@@ -11,57 +11,57 @@ require("scripts/zones/Port_Jeuno/TextIDs");
-	                if(count == 3 + Artifact_Armor_Plus_one[nb][6])then  --check the total number of item trade ( base af + af-1 + craft item + number of curency)
-                	AvaibleCombinaisonDectect = Artifact_Armor_Plus_one[nb-1];
-				    printf("detect trade  avaible Artifact combinaison: %u",Artifact_Armor_Plus_one[nb][1]);		
-				   end
-               end
-           end  
-     end
-	 
-   if(trade:hasItemQty(1875,count) and AvaibleCombinaisonDectect == 0)then  --- AB storage
-           local total = StoreAncientBeastcoins+count;
-           player:startEvent(0x0137,count,0,0,0,0,0,0,total);
-	          if(total < 9999)then  -- store max 9999 Ancien beastcoin
-                  player:setVar("Ancien_Beastcoin_store",total);	
-	              player:tradeComplete();
-	          end	   
+    local count = trade:getItemCount();
+    local CurrentAFupgrade = player:getVar("AFupgrade");
+    local StoreAncientBeastcoins = player:getVar("Ancien_Beastcoin_store");
+    local AvaibleCombinaisonDectect = 0;
+    local cost = 0;
+    local reliccost = 0;
+    local time = os.date("*t");
 
-   elseif(AvaibleCombinaisonDectect ~= 0)then
-     player:setVar("AFupgrade",AvaibleCombinaisonDectect);
-	 player:setVar("AFupgradeDay", tonumber(os.date("%j")) );
-	 reliccost = StoreAncientBeastcoins-cost;
-	 player:setVar("Ancien_Beastcoin_store",reliccost);	 
-	 player:tradeComplete();
-     player:startEvent(0x0138);
-   end  
+    if (CurrentAFupgrade == 0 and count == 4) then -- RELIC Armor +1 ???
+        for nb = 2, table.getn(Relic_Armor_Plus_one), 2 do  --looking for the relic armor
+            --trade base relic armor                                                                                    --trade temenos Item                                       --trade appolyion item                                                          --trade craft item                                                                                     have enought ancien beastcoin ctore ?
+            if (trade:hasItemQty(Relic_Armor_Plus_one[nb][2],1) and  trade:hasItemQty(Relic_Armor_Plus_one[nb][3],1) and  trade:hasItemQty(Relic_Armor_Plus_one[nb][4],1)  and  trade:hasItemQty(Relic_Armor_Plus_one[nb][5],1) and Relic_Armor_Plus_one[nb][6] <= StoreAncientBeastcoins) then 
+                AvaibleCombinaisonDectect = Relic_Armor_Plus_one[nb-1];
+                cost = Relic_Armor_Plus_one[nb][6];
+                printf("detect trade  avaible relic combinaison: %u", Relic_Armor_Plus_one[nb][1]);   
+            end
+        end
+    elseif (CurrentAFupgrade == 0 and AvaibleCombinaisonDectect == 0) then -- Artfact Armor +1 ???
+        for nb = 2, table.getn(Artifact_Armor_Plus_one), 2 do  --looking for the Artifact armor
+            --trade base Artfact armor                                                                  --- trade Artfact armor -1                                                                       trade craft item

```

Comment:
Since you're here anyways, Sagheera needs to use getCurrency instead of her
own player var